src/components/custom-coverage-date/custom-coverage-date.js

```diff
@@ -0,0 +1,169 @@
+      isEditing: false
+    });
+    if (this.props.pristine) {
+      return;
+    }
+    this.props.onSubmit(data);
+  }
+
+
+  render() {
+    let { pristine } = this.props;
+
+    const { beginCoverage, endCoverage } = this.props.initialValues;
+    if (this.state.isEditing) {
+      return (
+        <form onSubmit={this.props.handleSubmit(this.handleSubmit)}>
+          <div
+            data-test-eholdings-package-details-custom-coverage-inputs
+            className={styles['custom-coverage-dates']}
+          >
+            <div data-test-eholdings-package-details-custom-begin-coverage className={styles['custom-coverage-date-picker']}>
+              <Field
+                name='beginCoverage'
+                component={this.renderDatepicker}
+                label="Start Date"
+              />
+            </div>
+            <div data-test-eholdings-package-details-custom-end-coverage className={styles['custom-coverage-date-picker']}>
+              <Field
+                name='endCoverage'
+                component={this.renderDatepicker}
+                label="End Date"
+              />
+            </div>
+            <div className={styles['custom-coverage-date-trash-bin']}>
+              {this.props.initialValues.beginCoverage && (
+                <IconButton icon="trashBin" onClick={this.handleDeleteCustomCoverage} size="small" />
+              )}
+            </div>
+          </div>
+          <div className={styles['custom-coverage-action-buttons']}>
+            <div data-test-eholdings-package-details-cancel-custom-coverage-button>
+              <Button hollow type="button" role="button" onClick={this.handleCancelCustomCoverage} buttonStyle="secondary">
+                Cancel
+              </Button>
+            </div>
+            <div data-test-eholdings-package-details-save-custom-coverage-button>
+              <Button disabled={pristine} type="submit" role="button" buttonStyle="primary">
+                Save
+              </Button>
+            </div>
+          </div>
+        </form>
+      );
+    } else if (beginCoverage) {
+      return (
+        <div className={styles['custom-coverage-date-display']}>
+          <div data-test-eholdings-package-details-custom-coverage-display className={styles['custom-coverage-dates']}>
+            {beginCoverage} - {endCoverage || 'Present'}
+          </div>
+          <div data-test-eholdings-package-details-edit-custom-coverage-button>
+            <IconButton icon="edit" onClick={this.handleEditCustomCoverage} />
+          </div>
+        </div>
+      );
+    } else {
+      return (
+
+        <div data-test-eholdings-package-details-custom-coverage-button>
+          <Button
+            hollow
+            buttonStyle="primary"
+            type="button"
+            onClick={this.handleEditCustomCoverage}
+          >
+            Add Custom Coverage
+          </Button>
+        </div>
+      );
+    }
+  }
+}
+
+function validate(values, props) {
```

Comment:
minor nit, but i'd move this above the `render`.  i'd also throw some comments throughout this file, as `redux-form` is not the most obvious thing in the world.

Reply:
I think it can be moved. it will have to be passed as { validate: this.validate } which may work. And then it can put into the class above render.

Reply:
How will we reference the validation function from within the class from the redux-form connect method at the end of the file?